completers/common/fnm_completer/cmd/root.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"github.com/carapace-sh/carapace"
+	"github.com/carapace-sh/carapace-bin/completers/common/fnm_completer/cmd/action"
+	"github.com/carapace-sh/carapace/pkg/style"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "fnm",
+	Short: "A fast and simple Node.js manager",
+	Long:  "https://github.com/Schniz/fnm",
+	Run:   func(*cobra.Command, []string) {},
+}
+
+func Execute() error {
+	return rootCmd.Execute()
+}
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	rootCmd.PersistentFlags().String("arch", "", "Override the architecture of the installed Node binary. Defaults to arch of fnm binary")
```

Comment:
Cobra has `PersistenFlags` that are passed on to each subcommand (which seems to be the case here).
Additional benefit is that they aren't shown on the subcommand when already provided for a parent command.

Using `addCommonFlags` is otherwise fine as approach for repetitive flags.